### PR TITLE
CATROID-768 "Become focus point with __ % horizontal and __ % vertica…

### PR DESCRIPTION
**Pull request by frit96**
https://jira.catrob.at/browse/CATROID-768
"Become focus point with __ % horizontal and __ % vertical flexibility" brick
Designed a new Brick which focuses on a sprite and then moves the camera based on the position of the sprite with a vertical and horizontal flexibility. This means if you have a screen with 1000 pixelwidth and you choose a horizontal flexibility of 50%, the camera begins to move when the sprite hits either the 250th pixel or the 750th pixel. All other sprites remain in there position and dont move.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java [hunk 1]
```diff
@@ -131,6 +132,9 @@
 	public WebConnectionHolder webConnectionHolder;
 
 	private List<Sprite> sprites;
+	private Sprite spriteToFocusOn = null;
```
Comment (blindguardian50) [inline catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java:135]: Please take all changes you made in the StageListener and put it into a seperateClass as the StageListener class already has lots of code in it. You can then create a new member of this class for the StageListener and there call the methods.

File: catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java [hunk 2]
```diff
@@ -588,6 +598,39 @@ public void render() {
...
 			makeTestPixels = false;
 		}
+
+		if (spriteToFocusOn != null) {
+			calculateCameraPositionForFocusedSprite();
+		}
+	}
+
+	@VisibleForTesting
+	public void calculateCameraPositionForFocusedSprite() {
```
Comment (blindguardian50) [inline catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java:608]: Please rename to:
```suggestion
	public void updateCameraPositionForFocusedSprite() {
```
as the word calculate does not imply any action only a calculation.

File: catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java [hunk 3]
```diff
@@ -787,6 +830,18 @@ private void setSchedulerStateForAllLooks(@ThreadScheduler.SchedulerState int st
 		}
 	}
 
+	public void setSpriteToFocusOn(Sprite spriteToFocusOn) {
```
Comment (blindguardian50) [inline catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java:833]: You won't need the setters if you extract your changes into a seperate kotlin class.

**Review by blindguardian50 — CHANGES_REQUESTED** [review]
Good Work! Please consider my suggested changes!

**Review by blindguardian50 — APPROVED** [review]
Nice Work!

**Comment by wslany** [comment]
Please rebase and resolve th conflicts.

**Comment by frit96** [comment]
I rebased and solved the conlicts

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**